src/plugins/explore/public/components/visualizations/bar_gauge/to_expression.ts

```diff
@@ -58,40 +60,58 @@ export const createBarGaugeSpec = (
         g1.map((d) => d[numericField]),
         styleOptions.valueCalculation
       );
-      if (calculate) {
-        maxNumber = Math.max(maxNumber, calculate);
-      }
-
       const isValidNumber =
         calculate !== undefined && typeof calculate === 'number' && !isNaN(calculate);
 
+      if (calculate) {
+        maxNumber = Math.max(maxNumber, isValidNumber ? calculate : -Infinity);
+        minNumber = Math.min(minNumber, isValidNumber ? calculate : Infinity);
+      }
+
       const displayValue = showDisplayValue(isValidNumber, selectedUnit, calculate);
       maxTextLength = Math.max(maxTextLength, String(displayValue).length);
       newRecord.push({
         ...g1[0],
-        [numericField]: calculate,
+        [numericField]: isValidNumber ? calculate : null,
         displayValue,
       });
     }
   }
 
-  const validThresholds = [
-    { value: styleOptions?.min ?? 0, color: styleOptions.thresholdOptions.baseColor } as Threshold,
-    ...(styleOptions.thresholdOptions.thresholds?.filter(
-      (t) =>
-        t.value <= maxNumber &&
-        t.value >= (styleOptions.min ?? 0) &&
-        t.value <= (styleOptions.max ?? Infinity)
-    ) ?? []),
-  ] as Threshold[];
+  // To support negative values, when all the values are above 0, take minBase as 0 to start a bar.
+  // Similarly, when all values are below 0, take maxBase as 0 to end a bar
+
+  const maxBase = styleOptions?.max ?? Math.max(maxNumber, 0);
+  const minBase = styleOptions?.min ?? Math.min(minNumber, 0);
+
+  const invalidCase = minNumber === Infinity || minBase >= maxBase || minBase > maxNumber;
```

Comment:
```suggestion
  const invalidCase = minBase > maxBase;
```
If I understood correctly, this is good enough